index.js

```diff
@@ -42,14 +44,30 @@ module.exports = function requireAll(options) {
         filter: filter,
         excludeDirs: excludeDirs,
         map: map,
-        resolve: resolve
+        resolve: resolve,
+        merge: merge
       });
 
     } else {
       var name = filterFile(file);
       if (!name) return;
 
-      modules[map(name, filepath)] = resolve(require(filepath));
+      var module = resolve(require(filepath));
+
+      if (merge) {
+        // console.log('module and is object', module, isObject(module))
+      if (isObject(module)) {
+        for (let key in module) {
+          modules[key] = module[key];
```

Comment:
Yea, I didn't realize when I made that comment you're doing not only merging of the file names but even of the object within the file at the first level. `map()` probably doesn't make sense for this line.